test/stdlib/CharacterProperties.swift

```diff
@@ -0,0 +1,191 @@
+  expectEqual(2, Character("੨").wholeNumberValue)
+  expectEqual(2, Character("૨").wholeNumberValue)
+  expectEqual(3, Character("³").wholeNumberValue)
+  expectEqual(4, Character("٤").wholeNumberValue)
+  expectEqual(4, Character("൪").wholeNumberValue)
+  expectEqual(4, Character("೪").wholeNumberValue)
+  expectEqual(4, Character("౪").wholeNumberValue)
+  expectEqual(4, Character("௪").wholeNumberValue)
+  expectEqual(4, Character("୪").wholeNumberValue)
+  expectEqual(4, Character("૪").wholeNumberValue)
+  expectEqual(4, Character("੪").wholeNumberValue)
+  expectEqual(4, Character("๔").wholeNumberValue)
+  expectEqual(4, Character("໔").wholeNumberValue)
+  expectEqual(5, Character("५").wholeNumberValue)
+  expectEqual(5, Character("༥").wholeNumberValue)
+  expectEqual(5, Character("፭").wholeNumberValue)
+  expectEqual(5, Character("᠕").wholeNumberValue)
+  expectEqual(5, Character("Ⅴ").wholeNumberValue) // Roman numeral
+  expectEqual(5, Character("𐌡").wholeNumberValue)
+  expectEqual(5, Character("߅").wholeNumberValue)
+  expectEqual(5, Character("᭕").wholeNumberValue)
+  expectEqual(5, Character("𝍤").wholeNumberValue)
+  expectEqual(5, Character("᮵").wholeNumberValue)
+  expectEqual(6, Character("六").wholeNumberValue)
+  expectEqual(6, Character("六").wholeNumberValue) // Compatibility
+  expectEqual(7, Character("𝟩").wholeNumberValue) // Math san-serif
+  expectEqual(7, Character("㈦").wholeNumberValue)
+  expectEqual(7, Character("㊆").wholeNumberValue)
+  expectEqual(7, Character("𑁭").wholeNumberValue)
+  expectEqual(8, Character("꧘").wholeNumberValue)
+  expectEqual(8, Character("᪈").wholeNumberValue)
+  expectEqual(8, Character("᪘").wholeNumberValue)
+  expectEqual(8, Character("꩘").wholeNumberValue)
+  expectEqual(9, Character("๙").wholeNumberValue)
+
+  expectEqual(18, Character("⒅").wholeNumberValue)
+  expectEqual(20, Character("⑳").wholeNumberValue)
+  expectEqual(20, Character("𐄑").wholeNumberValue)
+  expectEqual(20, Character("𐏔").wholeNumberValue)
+  expectEqual(20, Character("𐤘").wholeNumberValue)
+  expectEqual(20, Character("〹").wholeNumberValue)
+  expectEqual(50, Character("ↆ").wholeNumberValue)
+  expectEqual(70, Character("𑁡").wholeNumberValue)
+  expectEqual(90, Character("𐍁").wholeNumberValue)
+  expectEqual(1_000, Character("𑁥").wholeNumberValue)
+  expectEqual(5_000, Character("ↁ").wholeNumberValue)
+  expectEqual(10_000, Character("万").wholeNumberValue)
+}
+
+CharacterPropertiesTests.test("Casing") {
+  let eAccent = Character("\u{0065}\u{0301}")
+  let EAccent = Character("\u{0045}\u{0301}")
+  expectTrue(eAccent.isLowercase && eAccent.isCased)
+  expectFalse(eAccent.isUppercase)
+  expectTrue(EAccent.isUppercase && EAccent.isCased)
+  expectFalse(EAccent.isLowercase)
+
+  expectTrue(Character("И").isUppercase)
+  expectTrue(Character("и").isLowercase)
+  expectTrue(Character("Π").isUppercase)
+  expectTrue(Character("π").isLowercase)
+
+  expectEqual("SS", Character("ß").uppercased())
```

Comment:
Is the [Capital ẞ](https://en.wikipedia.org/wiki/Capital_ẞ) (U+1E9E: LATIN CAPITAL LETTER SHARP S) supported by ICU yet?

```
expectEqual("ß", Character("ẞ").lowercased())
```

The system font (SF Pro) displays it as an "SS" ligature.